File: rest-common/src/main/java/org/chronopolis/rest/models/serializers/ZonedDateTimeDeserializer.java
package org.chronopolis.rest.models.serializers;

import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.JsonDeserializer;

import java.io.IOException;
import java.time.ZoneOffset;
import java.time.ZonedDateTime;
import java.time.format.DateTimeFormatter;

/**
 * Jackson deserializer for ZonedDateTime objects
 *
 * Created by shake on 12/16/16.
 */
public class ZonedDateTimeDeserializer extends JsonDeserializer<ZonedDateTime> {
    @Override
    public ZonedDateTime deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException {
        DateTimeFormatter fmt = DateTimeFormatter.ISO_OFFSET_DATE_TIME.withZone(ZoneOffset.UTC);
        String text = jsonParser.getText();
        return ZonedDateTime.from(fmt.parse(text));
    }
}

File: rest-models/src/main/kotlin/org/chronopolis/rest/kot/models/serializers/ZonedDateTimeDeserializer.kt
package org.chronopolis.rest.kot.models.serializers

import com.fasterxml.jackson.core.JsonParser
import com.fasterxml.jackson.databind.DeserializationContext
import com.fasterxml.jackson.databind.JsonDeserializer
import java.time.ZoneOffset
import java.time.ZonedDateTime
import java.time.format.DateTimeFormatter.ISO_OFFSET_DATE_TIME

class ZonedDateTimeDeserializer : JsonDeserializer<ZonedDateTime>() {

    override fun deserialize(jsonParser: JsonParser,
                             deserializationContext: DeserializationContext): ZonedDateTime {
        val text = jsonParser.text
        return ZonedDateTime.from(ISO_OFFSET_DATE_TIME.withZone(ZoneOffset.UTC).parse(text))
    }
}

